### ISSUE TITLE
Should filtering in the effect select dialog use substrings instead of regexes?

### ISSUE THREAD

**Issue by jack126guy** [LMMS/lmms#2495]
I ran into this while I was trying to add a CAPS (C*) effect to an instrument. The entry:

    c*

doesn't give what you might expect; you have to enter:

    c\*

to get those effects.

I don't know if everyone is familiar enough with regexes to figure this out, so I think the filtering should be done with substrings. (Perhaps there could be an option to enable regexes?)

I notice that `QRegExp` [can be changed to use fixed substrings](http://doc.qt.io/qt-4.8/qregexp.html#setPatternSyntax); perhaps that could be considered.

**Comment by softrabbit** [comment]
I agree, substrings would be better. You can't expect the general population to know regexes. Maybe "standard" wildcards (* and ? are pretty widespread) could work if something more than plain substrings is necessary.

**Comment by tresf** [comment]
> I notice that `QRegExp` can be [changed to use fixed substrings](http://doc.qt.io/qt-4.8/qregexp.html#setPatternSyntax); perhaps that could be considered.

I'm not sure that will work because from what I'm seeing, we're not using `QRegExp`, but rather [a `QString` from the change signal](https://github.com/LMMS/lmms/blob/master/src/gui/EffectSelectDialog.cpp#L99:L100).  What you are suggesting would assume we wrap the signal into a `QRegExp` each time a letter is typed so that we can manipulate the `QRegExp` behavior just-in-time (right?)

I find this issue particularly interesting because it illustrates a few user-interface problems.

1. We have no way of selecting plugins by provider, forcing users to try to search based on some obscure wildcard.  Although CAPS are popular to our users, asterisks in nomenclature generally aren't, but regardless, this issue exposes the need to select plugins by provider, which I've found myself wanting to do over time as well, probably worthy of an enhancement request.

2. This happens to be a rather rare collision since your wildcard also contains a wildcard.  In my tenure with LMMS, this is the first time I've seen someone try to locate a plugin with an asterisk in the name, so I would also question the usefulness of the prefix `C*` as a plugin name.  It appears half of the calf plugins start with the word `Calf` whereas the other half start with `C*`.  This seems like a bit of a code consistency problem, unless there's more meaning to the `C*` that I don't know about.

3.  An ugly hack for this instance would be to escape a list of wildcards just-in-time.  It would have scalability problems, but would be a fast fix for common wildcards like the asterisk.

4.  Another change worth considering is adjusting the offending Ladspa plugin names to to instead display `LADSPA: Calf ` if we agree that is preferred over the existing behavior.  This wouldn't fix the underlying problem, but would instead be a separate UX enhancement.

TL;DR though, the bug is valid and should be addressed.

**Comment by jack126guy** [comment]
> What you are suggesting would assume we wrap the signal into a QRegExp each time a letter is typed so that we can manipulate the QRegExp behavior just-in-time (right?)

I was going for that, but I found a much simpler solution: [`setFilterFixedString`](http://doc.qt.io/qt-4.8/qsortfilterproxymodel.html#setFilterFixedString). So we can just change the slot.

> It appears half of the calf plugins start with the word Calf whereas the other half start with C*. 

Calf? I thought those were a separate set from CAPS (a.k.a. C*).

**Comment by tresf** [comment]
> I was going for that, but I found a much simpler solution: `setFilterFixedString`. So we can just change the slot.

:+1: 

> Calf? I thought those were a separate set from CAPS (a.k.a. C*).

My mistake. :+1:
